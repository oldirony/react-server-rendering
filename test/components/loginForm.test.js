import React from 'react';
import LoginForm from '../../src/components/login-form/login-form';

const props = {
	handleSubmit: func => func.bind(null, {})
}

test('It renders correcly', ()=>{
	let form = shallow(<LoginForm {...props}/>);
	expect(form).toMatchSnapshot();

	form = shallow(<LoginForm {...props} authError="Error message"/>);
	expect(form).toMatchSnapshot();
})