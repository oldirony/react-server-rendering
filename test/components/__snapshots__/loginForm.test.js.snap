// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders correcly 1`] = `
<form
  onSubmit={[Function]}
>
  <h1>
    Login
  </h1>
  <div
    className="section"
  >
    <div
      className="row"
    >
      <Field
        className="input"
        component="input"
        name="email"
        placeholder="Email"
        type="email"
      />
    </div>
    <div
      className="row"
    >
      <Field
        className="input"
        component="input"
        name="password"
        placeholder="Password"
        type="password"
      />
    </div>
  </div>
  <div
    className="section"
  >
    <div
      className="row"
    >
      <button
        className="button fullwidth"
        type="submit"
      >
        Login
      </button>
    </div>
    <Link
      className="vanilla"
      replace={false}
      to="/signup"
    >
      Are you a new user?
    </Link>
  </div>
</form>
`;

exports[`It renders correcly 2`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="authError"
  >
    Error message
  </div>
  <h1>
    Login
  </h1>
  <div
    className="section"
  >
    <div
      className="row"
    >
      <Field
        className="input"
        component="input"
        name="email"
        placeholder="Email"
        type="email"
      />
    </div>
    <div
      className="row"
    >
      <Field
        className="input"
        component="input"
        name="password"
        placeholder="Password"
        type="password"
      />
    </div>
  </div>
  <div
    className="section"
  >
    <div
      className="row"
    >
      <button
        className="button fullwidth"
        type="submit"
      >
        Login
      </button>
    </div>
    <Link
      className="vanilla"
      replace={false}
      to="/signup"
    >
      Are you a new user?
    </Link>
  </div>
</form>
`;
